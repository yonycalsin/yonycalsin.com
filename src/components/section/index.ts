export * from './section'
export * from './section-header'
