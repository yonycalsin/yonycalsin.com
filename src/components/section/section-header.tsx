import * as React from 'react'
import { Box, Heading } from '@chakra-ui/react'

export interface SectionHeaderProps {
  title: string
  actionHref?: string
  actionLabel?: string
  actionHrefExternal?: boolean
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  actionComponent?: any
  children?: React.ReactNode
}

export function SectionHeader(props: SectionHeaderProps) {
  const {
    title,
    actionHref,
    actionLabel,
    actionComponent: ActionComponent = 'a',
    actionHrefExternal = false,
    children,
  } = props

  return (
    <div>
      <Box display="flex" alignItems="center" justifyContent="space-between" position="relative">
        <Heading size="lg" fontWeight="bold">
          {title}
        </Heading>
        {actionHref ? (
          <ActionComponent href={actionHref} target={actionHrefExternal ? '_blank' : null}>
            {actionLabel}
          </ActionComponent>
        ) : null}
      </Box>
      {children}
    </div>
  )
}
