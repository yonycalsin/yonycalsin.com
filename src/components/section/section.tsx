import type { SectionProps } from 'typings/components'

function Section(props: SectionProps) {
  const { children } = props

  return <section className="mt-6 md:mt-10">{children}</section>
}

export default Section
