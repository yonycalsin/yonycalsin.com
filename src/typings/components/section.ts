interface SectionProps {
  children: React.ReactNode
}

interface SectionHeaderProps {
  title: string
  className?: string
  actionHref?: string
  actionLabel?: string
  actionHrefExternal?: boolean
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  actionComponent?: any
  children?: React.ReactNode
}

export type { SectionHeaderProps, SectionProps }
