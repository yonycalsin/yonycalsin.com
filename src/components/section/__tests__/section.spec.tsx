import * as React from 'react'
import { render } from '@testing-library/react'

import { Section } from '../section'

const setup = () => {
  return render(
    <Section>
      <div>Section Content</div>
    </Section>,
  )
}

describe('Components / Section', () => {
  it('renders an section', () => {
    const view = setup()

    expect(view.baseElement).toMatchSnapshot()
  })
})
