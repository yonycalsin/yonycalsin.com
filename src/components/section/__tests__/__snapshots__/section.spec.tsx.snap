// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Section renders an section 1`] = `
<body>
  <div>
    <section
      class="mt-6 md:mt-10"
    >
      <div>
        Section Content
      </div>
    </section>
  </div>
</body>
`;
